rules/TOOL_trace_nsiqcppstyle_callbacks.py

```diff
@@ -0,0 +1,80 @@
+from nsiqcppstyle_rulemanager import *
+
+def PrintContextStack(contextStack):
+    if (contextStack != None) and (contextStack.contextstack != None) and (len(contextStack.contextstack) > 0):
+        indent_string = (20 * ' ')
+        print("%sContext stack (%d):" % (indent_string, len(contextStack.contextstack)))
+        for t in contextStack.contextstack:
+            print('%s    %s' % (indent_string, str(t)))
+
+def FunctionScopeRule(lexer, contextStack):
+    print("FunctionScopeRule (lexer, contextStack)")
+    PrintContextStack(contextStack)
+
+def FunctionNameRule(lexer, fullName, decl, contextStack, context):
+    print("FunctionNameRule  (lexer,")
+    print("                   fullName='%s'," % (fullName))
```

Comment:
IMHO, inserting the spaces lets me visually know that everything is lined up correctly (e.g., the `fullName` parameter display is one column past the function opening left parenthesis of the line above).  Complicating the visual code with formatting to avoid inserting spaces seems to be overkill to me.